code/datums/components/squish.dm

```diff
@@ -19,6 +19,7 @@
 	var/mob/living/carbon/C = target
 	var/was_lying = C.body_position == LYING_DOWN
 	addtimer(CALLBACK(src, PROC_REF(Detach), C, was_lying, reverse), duration)
+	ADD_TRAIT(target, TRAIT_FLATTENED, "vendor_flattening")
```

Comment:
I don't think applying it from the component is the best way to go about it. Squishes can stack, after all, and this would mean that you could knock a vendor onto yourself pre-emptively to become immune to vendor shoves for 80 seconds.

Might be better to have it as a status effect with a fairly short lifespan, say 10-30 seconds or so, and set it when the squish component gets applied? That way you can't be chainstunned, but can be pushed into more than one vending machine during a drawn-out fight.

Reply:
> you could knock a vendor onto yourself pre-emptively to become immune to vendor shoves for 80 seconds.

I mean like you're probably gonna end up taking anywhere from 60-100 brute damage and suffer IB, so seems like a rather shit deal to me at least!
Although status effect may be a good idea yeah, trait just seemed like a clean way to do it